api/app.py

```diff
@@ -38,19 +34,48 @@ def get_algorithms():
 @app.route('/analyze', methods=['POST'])
 @cross_origin(origin=['*'],supports_credentials=True)
 def analyze():
-    # export platform info to prometheus
-    response = requests.post(qujata_platform_exporter_target + "/export-platform-info")
-
     global test_is_running
-    print('analysis route was triggered.')
-    start_time = int(datetime.timestamp(datetime.now() - timedelta(seconds=60)) * 1000)
     data = request.get_json()
-    print(test_is_running)
+
+    __export_platform_data()
+    error = __validate(data)
+    if error != None:
+        return error
+
+    test_is_running = True
+    start_time = int(datetime.timestamp(datetime.now() - timedelta(seconds=60)) * 1000)
+    error = __start_analyze(data)
+    if error != None:
+        return error
+    end_time = int(datetime.timestamp(datetime.now() + timedelta(seconds=90)) * 1000)
+    test_is_running = False
+
+    run_id = str(uuid.uuid4())
+    
+    return jsonify({
+        'run_id': run_id,
+        'from': start_time,
+        'to': end_time
+    })
+
+def __export_platform_data():
+    requests.post(qujata_platform_exporter_target + "/export-platform-info")
+
+
+def __validate(data):
+    global test_is_running
     if not data or 'algorithms' not in data:
-        return jsonify({'error': 'Invalid data provided'}), 400
+        return jsonify({'error': 'Invalid data provided', 'message': 'missing algorithms'}), 400
+    if data['iterationsCount'] < 500 or data['iterationsCount'] > 100000:
+        return jsonify({'error': 'Invalid data provided', 'message': 'iterationsCount must be greater then 500 and less then 100000'}), 400
     if test_is_running:
-        return jsonify({'error': 'Analisys test is running now'}), 409
-    test_is_running = True
+        return jsonify({'error': 'Current test is still running', 'message':'The previous test is still running. Please try again in few minutes'}), 423
```

Comment:
423 is very accurate for this error, nice! 
consider using "Constants" for all HTTP codes
For example

LOCKED = 423
BAD_REQUEST = 400